### PR TITLE
feat: Updated the design of the article page

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR includes,
1. Updated the design
2. the 'mark as code' styling (adding backticks in preview)

Fixes 
https://linear.app/chatwoot/issue/DES-18/article-content-editor
https://linear.app/chatwoot/issue/CW-2655/fix-issues-in-article-content-editor
https://linear.app/chatwoot/issue/CW-2623/update-the-design-of-the-article-page

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/public/api/v1/portals/articles/_article_header.html.erb [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
+<div class="flex items-center flex-row mb-6 gap-px">
+  <a
+    class="text-slate-500 dark:text-slate-200 text-sm gap-1 hover:cursor-pointer <%= @is_plain_layout_enabled && 'hover:underline' %> leading-8 font-semibold"
+    href="/hc/<%= @portal.slug %>/<%= @article.category&.locale %><%= @theme.present? && @theme != 'system' ? '?theme='+@theme : ''  %>"
+  >
+    <%= I18n.t('public_portal.common.home') %>
+  </a>
+  <span class="w-4 h-4 [&>svg]:w-3 [&>svg]:h-3 flex items-center justify-center text-xs text-slate-500 dark:text-slate-300"><%= render partial: 'icons/chevron-right' %></span>
+  <% if @article.category %>
+    <a class="text-slate-500 dark:text-slate-200 text-sm gap-1 whitespace-nowrap hover:cursor-pointer <%= @is_plain_layout_enabled && 'hover:underline' %> leading-8 font-semibold" href="/hc/<%= @portal.slug %>/<%= @article.category.locale %>/categories/<%= @article.category.slug %><%= @theme.present? && @theme != 'system' ? '?theme='+@theme : ''  %>">
```
Comment (nithindavid) [inline app/views/public/api/v1/portals/articles/_article_header.html.erb:10]: @sojan-official Can we do anything for links like this? We are repeating these at many places

File: app/views/public/api/v1/portals/articles/_article_header.html.erb [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
+    <%= @article.category&.name %>
+    </a>
+    <span class="w-4 h-4 [&>svg]:w-3 [&>svg]:h-3 flex items-center justify-center text-xs text-slate-500 dark:text-slate-300"><%= render partial: 'icons/chevron-right' %></span>
+    <span class="text-sm text-slate-800 dark:text-slate-100 font-semibold overflow-hidden text-ellipsis min-w-0 whitespace-nowrap"><%= @article.title %></span>
+  <% else %>
+    <span class="text-slate-700 dark:text-slate-100 leading-8 text-sm font-semibold" > Uncategorized </span>
```
Comment (nithindavid) [inline app/views/public/api/v1/portals/articles/_article_header.html.erb:16]: i18n

**Review by nithindavid — CHANGES_REQUESTED** [review]
Minor changes needed. Please see if we can improve category sub text as well.

File: app/views/public/api/v1/portals/articles/_article_header.html.erb (hunk 1 above)
Comment (iamsivin) [reply app/views/public/api/v1/portals/articles/_article_header.html.erb:10]: @nithindavid It's handled in the last PR https://github.com/chatwoot/chatwoot/pull/8230/files#diff-507eaef3ab83df007fe2b82a5364439aebd3fa8fa2ef5234eaf5bcd050754447R47 . Now we have three helpers for opening links to the home page, category page, and article page.

File: app/javascript/portal/components/TableOfContents.vue [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
-    <div v-if="rows.length > 0" class="sticky top-24 py-12 overflow-auto">
+  <div class="hidden lg:block flex-1 py-6 scroll-mt-24 pl-4">
+    <div v-if="rows.length > 0" class="sticky top-24 py-2 overflow-auto">
       <nav class="max-w-2xl">
         <h2
           id="on-this-page-title"
-          class="text-slate-800 dark:text-slate-50 font-semibold tracking-wide border-b mb-3 leading-7"
+          class="text-slate-800 pl-6 dark:text-slate-50 font-semibold tracking-wide py-3 leading-7 border-l-2 border-solid border-slate-100 dark:border-slate-800"
+          :class="{
+            '!border-slate-400 dark:!border-slate-100': !currentSlug,
```
Comment (scmmishra) [inline app/javascript/portal/components/TableOfContents.vue:9]: Lets not use the important tag here. We can also conditionally add the default `border-slate-100 dark:border-slate-800`

File: app/javascript/portal/components/TableOfContents.vue [hunk 4]
```diff
@@ -65,6 +85,30 @@ export default {
 
       return '';
     },
+    onURLHashChange() {
+      this.currentSlug = window.location?.hash?.substring(1) || '';
+    },
+    isElementActive(el) {
+      return this.currentSlug === el.slug;
+    },
+    tocHeaderBorderStyles() {
```
Comment (scmmishra) [inline app/javascript/portal/components/TableOfContents.vue:94]: This isn't used anywhere it seems

File: app/views/icons/_chatwoot-logo.html.erb
Comment (scmmishra) [inline app/views/icons/_chatwoot-logo.html.erb:1]: This is a base64 image inside an SVG, see if we can use a SVG only image. The payload will be too huge wherever you add this

**Review by scmmishra — CHANGES_REQUESTED** [review]

**Pull request merged**